server/src/main/java/org/opensearch/index/cache/request/ShardRequestCache.java

```diff
@@ -76,6 +79,15 @@ public void onRemoval(long keyRamBytesUsed, BytesReference value, boolean evicte
             dec += value.ramBytesUsed();
         }
         totalMetric.dec(dec);
+        if (totalMetric.count() < 0) {
+            totalMetric.inc(dec);
+            logger.warn(
+                "Ignoring the operation to deduct memory: {} from RequestStats memory_size metric as it will "
+                    + "go negative. Current memory: {}. This is a bug.",
+                dec,
+                totalMetric.count()
+            );
```

Comment:
logger.error?

Reply:
I think this should be `logger.warn` considering this isn't preventing the domain to function normally, it would just have its requestStats miscalculated.
But looking for more opinions on this. @msfroh What do you think?

Reply:
Yeah -- I agree. I tend to reserve `error` for "I failed to do a thing", whereas I think of `warn` as "I'm doing a thing, but poorly".

Reply:
I still see there is a race condition here as multiple threads can deduct the bytes from totalMetric and then increment can be done for other thread. What are we trying to achieve by incrementing it back here ?

Reply:
@sohami We just don't wanted this to go negative and break stats response for now. We will eventually remove this logic once we don't see negative stats issue again in the long run.